### DISCUSSION TITLE
Only accept connections from specific node type

### DISCUSSION THREAD

**Discussion by FreshLucas-git** [bcakmakoglu/vue-flow#887] [Q&A] [answered]
Hi. Can we set a node that only allows connections from a specific source node? For example, Node A and Node B both connect to the Node C in the following image, but I want Node C to only accept connections from Node A. I'm wondering if we can limit a condition in Node C that only accepts connections from a specific source node such as Node A, so that other nodes cannot connect to it.

![example](https://user-images.githubusercontent.com/65151316/233964812-92556ae4-cb45-47d5-aced-e9f3dd375c49.png)

**Comment by bcakmakoglu** [comment]
You can use the `isValidConnection` prop of handles to validate connections.
[There's also an example in the docs.](https://vueflow.dev/examples/edges/validation.html)

**Reply by FreshLucas-git** [reply]
Hi @bcakmakoglu. Thank you for quick reply. I checked this example before. This scenario of example is slight different from my scenario. I hope that I just set one limitation in Node C, instead of using `isValidConnection` in Node A and Node B. Is that possible to implement it?

**Reply by bcakmakoglu** [reply, marked as answer]
Validation always happens from the source handle, not the target handle, so you *have to* pass `isValidConnection` to whatever source handle you're using at the moment.

Alternatively you can just allow the connection and catch it in `onConnect` and if you're not happy with the potential connection, just don't create a new edge.
